srcts/src/utils/object.ts

```diff
@@ -1,5 +1,10 @@
-function hasOwnProperty(x: { [key: string]: unknown }, y: string): boolean {
-  return Object.prototype.hasOwnProperty.call(x, y);
+// Inspriation from https://fettblog.eu/typescript-hasownproperty/
+// But mixing with "NonNullable key of Obj" instead of "key to unknown values"
+function hasOwnProperty<Prop extends keyof X, X extends { [key: string]: any }>(
+  obj: X,
+  prop: Prop
+): obj is X & { [key in NonNullable<Prop>]: X[key] } {
+  return Object.prototype.hasOwnProperty.call(obj, prop);
```

Comment:
This method has been the bane of this PR. 

The original type definition did not allow for type narrowing as it only returned a boolean value. I could not find another implementation that resolved my type narrowing needs without adding `{[key: string]: unknown}` which _poisoned_ the original type value for the remainder of the method. 

Thank you @nstrayer for the brainstorming session!